src/hooks/useIsAppleDevice.ts

```diff
@@ -0,0 +1,24 @@
+import { useEffect, useState } from 'react';
+
+export const useIsAppleDevice = () => {
+    const [isAppleDevice, setIsAppleDevice] = useState<boolean>();
+
+    useEffect(() => {
+        const platform =
+            (
+                navigator as unknown as {
+                    userAgentData: { platform: string };
+                }
+            )?.userAgentData?.platform ||
+            navigator?.platform ||
+            'unknown';
+
+        setIsAppleDevice(
+            platform.includes('Mac') ||
```

Comment:
I think we need to make a same-case check here, because this is not passing on my Mac, this value is `macOS` for me:
```suggestion
            platform.toLowerCase().includes('mac') ||
```